### PR TITLE
Annotated tests for multi_mx_call.

### PR DESCRIPTION
**Pull request by pykello**

### CODE REVIEW

File: src/test/regress/expected/multi_mx_call.out [hunk 1]
```diff
@@ -2,22 +2,38 @@
 -- Create worker-local tables to test procedure calls were routed
 set citus.shard_replication_factor to 1;
 set citus.replication_model to 'streaming';
-create table mx_call_dist_table(id int, val int);
-select create_distributed_table('mx_call_dist_table', 'id');
+--
+-- Utility UDFs
+--
+-- 1. Marks the given procedure as collocated with the given table.
```
Comment (serprex) [inline src/test/regress/expected/multi_mx_call.out]: colocated

File: src/test/regress/expected/multi_mx_call.out [hunk 2]
```diff
@@ -27,366 +43,252 @@ select create_distributed_table('mx_call_dist_table_enum', 'key');
...
     y := x + (select case groupid when 0 then 1 else 0 end from pg_dist_local_group);
-    y := y + (select sum(t1.val + t2.val) from mx_call_dist_table t1 join mx_call_dist_table2 t2 on t1.id = t2.id);
+    -- we also make sure that we can run distributed queries in the procedures
+    -- that are routed to the workers.
+    y := y + (select sum(t1.val + t2.val) from mx_call_dist_table_1 t1 join mx_call_dist_table_2 t2 on t1.id = t2.id);
 END;$$;
-CREATE PROCEDURE mx_call_proc_asdf(INOUT x mx_call_enum, INOUT y mx_call_enum) LANGUAGE plpgsql AS $$
+-- create another procedure which verifies:
+-- 1. we work fine with multiple return columns
+-- 2. we work find in combinaiton with custom types
```
Comment (serprex) [inline src/test/regress/expected/multi_mx_call.out]: combination

Comment (pykello) [reply src/test/regress/expected/multi_mx_call.out]: Fixed in the sql file, upstream is failing to build, so cannot generate the out file.

**Comment by serprex** [comment]
This got merged in a round about way, https://github.com/citusdata/citus/commit/dea26523b31c461ea389b4ab527e0bd56125e778

**Pull request closed without merging**